From bb9f9ffbb151397545f921cee5b6a4933c6eea80 Mon Sep 17 00:00:00 2001
Message-Id: <bb9f9ffbb151397545f921cee5b6a4933c6eea80.1666169374.git.stefan@agner.ch>
From: John Wolfe <jwolfe@vmware.com>
Date: Wed, 10 Aug 2022 06:12:02 -0700
Subject: [PATCH] Properly check authorization on incoming guestOps requests

Fix public pipe request checks.  Only a SessionRequest type should
be accepted on the public pipe.

Upstream: https://github.com/vmware/open-vm-tools/blob/CVE-2022-31676.patch/1205-Properly-check-authorization-on-incoming-guestOps-re.patch
Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 vgauth/serviceImpl/proto.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/vgauth/serviceImpl/proto.c b/vgauth/serviceImpl/proto.c
index db7159ee..c4f85b02 100644
--- a/vgauth/serviceImpl/proto.c
+++ b/vgauth/serviceImpl/proto.c
@@ -1,5 +1,5 @@
 /*********************************************************
- * Copyright (C) 2011-2016,2019-2021 VMware, Inc. All rights reserved.
+ * Copyright (c) 2011-2016,2019-2022 VMware, Inc. All rights reserved.
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU Lesser General Public License as published
@@ -1201,6 +1201,10 @@ Proto_SecurityCheckRequest(ServiceConnection *conn,
    VGAuthError err;
    gboolean isSecure = ServiceNetworkIsConnectionPrivateSuperUser(conn);
 
+   if (conn->isPublic && req->reqType != PROTO_REQUEST_SESSION_REQ) {
+      return VGAUTH_E_PERMISSION_DENIED;
+   }
+
    switch (req->reqType) {
       /*
        * This comes over the public connection; alwsys let it through.
-- 
2.38.0

